### PR TITLE
Feat/sb withdrawals

### PR DESCRIPTION
**Pull request by plondon**
## Description (optional)
Add a concise explanation of the changes.

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendEth/sagaRegister.ts [hunk 1]
```diff
@@ -21,11 +21,12 @@ export default ({ api, coreSagas, networks }) => {
       AT.SEND_ETH_SECOND_STEP_SUBMIT_CLICKED,
       sendEthSagas.secondStepSubmitClicked
     )
-    yield takeLatest(actionTypes.CHANGE, sendEthSagas.formChanged)
     yield takeLatest(
+      // @ts-ignore
```
Comment (plondon) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendEth/sagaRegister.ts:25]: Was causing ts compiler issues for some unknown reason

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts [hunk 2]
```diff
@@ -177,31 +190,65 @@ export default ({ coreSagas, networks }) => {
...
+              {
+                coin: payload
+              }
+            )
           )
           break
         case 'from':
-          const fromType = prop('type', payload)
-          if (is(String, payload)) {
+          let payloadT = payload as BtcFromType
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts:202]: `const`?

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts [hunk 3]
```diff
@@ -177,31 +190,65 @@ export default ({ coreSagas, networks }) => {
...
-            case ADDRESS_TYPES.LOCKBOX:
-              payment = yield payment.from(payload.xpub, fromType)
+            case 'CUSTODIAL':
+              const appState = yield select(identity)
+              const currency = selectors.core.settings
+                .getCurrency(appState)
+                .getOrFail('Can not retrieve currency.')
+              const bchRates = selectors.core.data.bch
+                .getRates(appState)
+                .getOrFail('Can not retrieve bitcoin rates.')
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendBch/sagas.ts:222]: BCH

File: packages/blockchain-wallet-v4-frontend/src/data/components/sendEth/sagas.ts [hunk 4]
```diff
@@ -100,60 +111,119 @@ export default ({ api, coreSagas, networks }) => {
...
           // After updating payment success check if to isContract
           yield put(A.sendEthCheckIsContract(value))
           return
         case 'amount':
-          const coinCode = prop('coinCode', payload)
+          const amountPayload = payload as SendEthFormAmountActionType['payload']
+          const coinCode = prop('coinCode', amountPayload)
           const weiAmount = Exchange.convertCoinToCoin({
-            value: prop('coin', payload),
+            baseToStandard: false,
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/sendEth/sagas.ts:212]: Intended?

File: packages/blockchain-wallet-v4/src/redux/payment/btc/types.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  address: string
+  label: string
+  type: 'LEGACY'
+  watchOnly: boolean
+}
+
+export type BtcFromType =
+  | BtcAccountFromType
+  | BtcCustodialFromType
+  | BtcLegacyFromType
```
Comment (schnogz) [inline packages/blockchain-wallet-v4/src/redux/payment/btc/types.ts:47]: much needed types! 🎉

File: packages/blockchain-wallet-v4/src/redux/payment/xlm/sagas.ts [hunk 6]
```diff
@@ -179,13 +190,13 @@ export default ({ api }) => {
...
 
-        const from = {
+        from = {
           type: fromType,
           address: accountId,
           account
         }
         const reserve = yield call(getReserve, accountId)
-        const effectiveBalance = yield call(
+        effectiveBalance = yield call(
```
Comment (schnogz) [inline packages/blockchain-wallet-v4/src/redux/payment/xlm/sagas.ts:199]: `const`?

**Comment by schnogz** [comment]
Going to need QA to test a wide swath of things.  But nice job, those new types are 🔥!

**Pull request merged**